### DISCUSSION TITLE
Have a config file

### DISCUSSION THREAD

**Discussion by ocimjr** [corrados/edrumulus#96] [Ideas]
When testing edrumulus I always have to mess with files like:

1. edrumulus_hardware.cpp to change pins
2. edrumulus.ino to change pad type and midi notes

I don't any ideas for how to implement this or even if it is feasible(optimisation) but, i feel it should have a file that we can change pins, pads and midi notes in the same place without seen the code :)
So, this should make config and code separate parts and potentially prevents some errors.

Maybe some constants like PIN_SNARE_HEAD , MIDI_SNARE_HEAD, MIDI_CHOKE_CRASH1 and so on...

And if you only configure for example PIN_SNARE_HEAD, MIDI_SNARE_HEAD, PAD_SNARE the code will adapt and only configure one pad. This should prevent some problems with array in edrumulus_hardware.cpp

Thank you again for you work, and hope this can be considared in a future implementation.

Octavio

**Comment by corrados** [comment]
> edrumulus.ino to change pad type and midi notes

The MIDI notes and pad types should not be set in the code and compiled on the board but should be set using the [edrumulus_gui.py](https://github.com/corrados/edrumulus/blob/main/tools/edrumulus_gui.py) tool (see [this screen shot](https://github.com/corrados/edrumulus/blob/main/doc/images/edrumulus_console_gui.jpg)). You'll need this tool later on anyway when you have to adjust your threshold, sensitivity, etc. values.

**Comment by corrados** [comment]
> edrumulus_hardware.cpp to change pins [...] i feel it should have a file that we can change pins, pads and midi notes in the same place without seen the code

The MIDI settings are already covered by my first comment. Regarding the pin definitions: Usually, Arduino projects let the user define the project settings in the .ino file. Having to make modifications in the edrumulus_hardware.cpp file is not good, I totally agree with you.

I have created a new Git branch [improve_hardware_config](https://github.com/corrados/edrumulus/tree/improve_hardware_config), where I moved the default pin definition in the header of the .ino file. With this modification, Edrumulus is compatible to other Arduino projects in this regard. Here is are the changes I have done: https://github.com/corrados/edrumulus/compare/main...improve_hardware_config

**Comment by ocimjr** [comment]
Cool, thanks I'll close this discussion
